### ISSUE TITLE
version (level 2) - version of topic hierarchy or of message format?

### ISSUE THREAD

**Issue by antje-s** [wmo-im/wis2-topic-hierarchy#11]
Currently the notes states that it would be "Alphabetic version of the topic hierarchy", but originally it was for the version of the message format - as far as I know. I think it would be good to use the version of message format, because then you have a possibility to introduce a new message format and all consumers can switch independently

**Comment by tomkralidis** [comment]
@antje-s is this issue related to #8?

**Comment by kaiwirt** [comment]
@tomkralidis Yes

**Comment by tomkralidis** [comment]
Thanks @kaiwirt.  Should we close this as a duplicate?

**Comment by kaiwirt** [comment]
I am not sure. They are related. #8 is the value that goes to the topic (a or v04). #11 is about what this value means. Is it the version of the topic hierarchy or the version of the message format. Or both?

I don't know if this is the same question.

**Comment by golfvert** [comment]
v04 is to designate the message format
a is for the topic structure

As both can change independently, it makes sense to have two different "version" number.
The confusion bit is that we use a (and not v01) and v04 (and not d). 

I think we should keep two different versioning (topic and message are unrelated). Should we unify the format?
Eg v01 and not a??

**Comment by tomkralidis** [comment]
Note that `version` is also in the notification message itself.

**Comment by antje-s** [comment]
Yes, there are two different versions (one for topic structure and one for the message format). If we include message format version in the topictree, we can switch to new message formats later and offer both formats in a transitional phase. Clients could bei sure to be able to handle messages received under the specific topic tree. 
If not included it could bei difficult to change message format for mandatory fields.

**Comment by antje-s** [comment]
Adjustments in the topic tree can be handled more easily by publishing messages to both topics in a transition phase.

**Comment by josusky** [comment]
If we ever need to change the format of the WNM, currently  `v04`, to an incompatible version `v05`, then it will be useful for the (legacy) clients to be able to identify end-points that offer WNM in the version they understand. That is why in `v03` the version was part of the topic. I see following alternatives, that do not require having WNM version as part of the topic:
* require the current (v04) clients to **silently** reject WNM that have higher version (e.g. `v05`) - during the transition period the clients would get notifications twice - once as `v04` and once as `v05`
* migration to new version (`v05`) would be done by having second set of brokers (including Global Brokers) with different host names or port numbers to differentiate them from endpoints that will provide `v04`.

Any other suggestions? Because to me both seems much more complicated then having `v0X` in the topic.

When it concerns the topic hierarchy, changes that rename a category within the TH may have serious impact on data consumers and will require migration period when both the old and new TH are used (and some WNM are published twice), and here again a version as part of the TH seems to me as the simplest solution. But alternatives similar to those mentioned above are possible too.

So we have two options for versioning in general (or perhaps three):
- If version in TH is the right solution for any changes that require a "migration period" then in addition to `a` (currently level 2) that denotes TH version, we should add `v04` to indicate WNM version (as an additional TH level) .
- If there is a better solution, then we can drop the version from TH completely, including the famous `a`.
- As a compromise we could have one identifier for any kind of compatibility-breaking changes be it in TH or WNM.

**Comment by kaiwirt** [comment]
In my opinion this issue can be closed.

**Comment by tomkralidis** [comment]
TT-WISMD 2028-08-28:
- no version bumps given subdomain changes (which will be frequent)
- could WTH have a WCMP2 record itself?
- could be communicated via a WIS2 operational newsletter
- given FT (notification of FT approval), published to operational newsletter and in manual on codes
- WTH will NOT be inline in manual on codes
- WIGOS updates example: links are in manual on codes

- what is a breaking change
  - removals
  - changes in structure
  - NOT additions
  - any rename
  - ONLY major changes to WNM
- subdomain updates should be considered case-by-case
- needs to be augmented with newsletter updates/similar communications

- brokers need to be aware of new centre ids
  - would be another workflow
